src/cromshell/status/command.py

```diff
@@ -0,0 +1,123 @@
+    if workflow_status_dic['status'] == 'Failed' or workflow_status_dic['status'] == 'Aborted' or workflow_status_dic[
+        'status'] == 'fail':
+        retVal = 1
+
+    if retVal == 1:
+        IOUtils.turtleDead(quiet)
+    elif workflow_status_dic['status'] == 'Succeeded':  # change to be Running
+        # Status claims this workflow is running fine, but we need to check to see
+        # if there are any failed sub-processes.
+        # To do this, we use the `execution-status-count` logic with some filtering:
+
+        tmpMetadata = tempfile.TemporaryFile()
+
+        # Get execution status count and filter the metadata down:
+        curl_meta_out = requests.get("{}/api/workflows/v1/{}/metadata?{}".format(cromwell_config.server, workflow_id,
+                                                                                 cromwell_config.slim_metadata_parameters))
+        tmpMetadata.write(curl_meta_out.content)
+
+        execution_status_count_list = jq.compile('.. | .calls? | values | map_values(group_by(.executionStatus) | '
+                                                 'map({(.[0].executionStatus): . | length}) | add)').input(
+            json.loads(curl_meta_out.content)).all()
+
+        # Check for failure states:
+        failed = 'false'
+        for dictionary in execution_status_count_list:
+            for key in dictionary:
+                if re.search('Failed', str(dictionary[key]), re.I):  # pattern should be Failed
+                    failed = 'true'
+                else:
+                    failed = 'false'
+
+        # Check for failures:
+        if failed == 'false':
+            # We could not find 'Fail' in our metadata, so our original Running status is correct.
+            IOUtils.turtle(quiet)
+        else:
+            IOUtils.doomedLogo()
+            workflow_status = 'DOOMED'
+            newtempfile = tempfile.TemporaryFile()
+            message = "The workflow is Running but one of the instances has failed, which will lead to failure"
+            templine = '{{\"status\":\"{}\",\"id\":\"{}\"}}{}{}'.format(workflow_status, workflow_id, "\n", message)
+            newtempfile.write(bytes(templine.encode()))
+            tempf = newtempfile
+    else:
+        IOUtils.turtle(quiet)
+
+    # Display status to user:
+    tempf.seek(0)
+    for line_binary in tempf.readlines():
+        line_string = line_binary.decode("utf-8")
+        print(line_string.replace(",", ",\n"))
+
+    # Update ${CROMWELL_SUBMISSIONS_FILE}:
+    with fileinput.FileInput(cromwell_config.submission_file, inplace=True, backup='.bak') as f:
+        pattern = '\t[^\t]+$'
+        replace = '\t' + workflow_status
+        for line in f:
+            if cromwell_config.server and workflow_id in line:
+                updated_line = re.sub(pattern, replace, line)
+                print(updated_line, end='\n')
+            else:
+                print(line, end='')
+
+    # closes and deletes temp file
+    tempf.close()
```

Comment:
if we go this in-memory solutions, we shouldn't need to worry about that here.

Reply:
done